xbmc/filesystem/posix/PosixFile.cpp

```diff
@@ -313,6 +314,18 @@ bool CPosixFile::Rename(const CURL& url, const CURL& urlnew)
   
   if (rename(name.c_str(), newName.c_str()) == 0)
     return true;
+
+  // rename across mount points - need to copy/delete
+  if (errno == EXDEV)
```

Comment:
Should be inserted after https://github.com/xbmc/xbmc/pull/6582/files#diff-35414b4f48bbb1c63412eba269ae96d9L317 of `errno` will be incorrect for later check.